services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/HankeServiceImpl.kt

```diff
@@ -271,39 +245,160 @@ class HankeServiceImpl(private val hankeRepository: HankeRepository,
+
+        // TODO: this method of removing entries if they are missing in the incoming data is different to
+        //     behavior of the other simpler fields, where missing or null field is considered "keep the existing value,
+        //     and return it back in response".
+        //     However, those simpler fields can not be removed as a whole, so they _can_ behave so.
+        //     For clarity, yhteystieto-entries should have separate action for removal (but then they should also
+        //     have separate action for addition, i.e. own API endpoint in controller).
+        //     So, consider the code below as "for now".
+        // If there is anything left in the existingYTs map, they have been removed in the incoming data,
+        // so remove them from the entity's list and make the back-reference null (and thus delete from the database).
+        // TODO: it should not be a list, but a "bag", or ensure it is e.g. linkedlist instead of arraylist (or similars).
+        //    The order of YTs does not matter(?), and removing things from e.g. array list
+        //    gets inefficient. Since there are so few entries, this crude solution works, for now.
+        entity.listOfHankeYhteystieto.removeAll(existingYTs.values)
+        for (hankeYht in existingYTs.values) {
+            hankeYht.hanke = null
+        }
+    }
+
+
+    private fun processIncomingHankeYhteystietosOfSpecificTypeToEntity(
+            listOfHankeYhteystiedot: List<HankeYhteystieto>, hankeEntity: HankeEntity, contactType: ContactType,
+            userid: Int, existingYTs: MutableMap<Int, HankeYhteystietoEntity>) {
+        for (hankeYht in listOfHankeYhteystiedot) {
+            var someFieldsSet = false
+            var validYT = false
+            // The UI has currently bigger problems implementing it so that YT entries that haven't even been touched
+            //   would not be sent to backend as a group of ""-fields, so the condition here is to make
+            //   such fully-empty entry considered as non-existent (i.e. skip it).
+            //   Also, for now, if anything is given, it is checked that all 4 mandatory fields are given, or we log an error and skip it.
+            // Note, validator should have enforced that at least the four main fields are all set (or all of them are empty).
+            // TODO: Currently validator does allow through an entry with four mandatory fields empty, but organisation field(s) non-empty!
+            // If anything field is given (not empty and not only whitespace)...
+            if (hankeYht.sukunimi.isNotBlank() || hankeYht.etunimi.isNotBlank()
+                    || hankeYht.email.isNotBlank() || hankeYht.puhelinnumero.isNotBlank()
+                    || !hankeYht.organisaatioNimi.isNullOrBlank() || !hankeYht.osasto.isNullOrBlank()) {
+                someFieldsSet = true
```

Comment:
This seems to do the trick. It might have been a place for a isYTValid() method even..? But consider yourself if you want to do that... This is hopefully temporary code.

Reply:
Yes, depending on how the controller validations and/or frontend handling of empty fields could change, these may still change. However, that "some field not empty" -situation is something we may need to be ready to handle always (one way or another). I'll add separate methods for both condition groups, make it more readable.